packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -2657,15 +2658,23 @@ export class ProjectService extends BaseService {
         };
         const schema = ProjectService.getWarehouseSchema(credentials);
 
-        const warehouseCatalog = warehouseClient.getFields(
+        const warehouseCatalog = await warehouseClient.getFields(
             tableName,
             schema,
             queryTags,
         );
 
         await sshTunnel.disconnect();
 
-        return warehouseCatalog;
+        if (!schema) {
+            throw new NotFoundError(
+                'Schema not found in warehouse credentials',
+            );
+        }
+
+        return warehouseCatalog[warehouseClient.credentials.type][schema][
+            tableName
+        ];
```

Comment:
I would like to return only an object with the fields for a table. Would this be the correct way of doing this? 
Do you think we shouldn't  return this format instead?
If we should, what do you think about the handling of undefined `schema`?